Early stopping doc (#3193)
PyTorchLightning/pytorch-lightning: docs/source/early_stopping.rst

There are two ways to enable the EarlyStopping callback:

-   Set `early_stop_callback=True`.
    If a dict is returned by
    :meth:`~pytorch_lightning.core.lightning.LightningModule.validation_epoch_end`,
    the callback will look for `val_loss` in the dict

>     and raise an error if `val_loss` is not present.

Reviewer: ```suggestion
    and display a warning if `val_loss` is not present.
```

    Otherwise, if a :class:`~pytorch_lightning.core.step_result.Result` is returned, by
    :meth:`~pytorch_lightning.core.lightning.LightningModule.validation_epoch_end`,
    :meth:`~pytorch_lightning.core.lightning.LightningModule.validation_step` or
    :meth:`~pytorch_lightning.core.lightning.LightningModule.training_step`,

>     the `early_stop_on` metric, specified in the initialization of the

Author: In that case early stopping is not triggered for me. I wanted to ask, if that is the expected behaviour or not after my other issue related to early stopping (#3199) is fixed. I didn't want to swamp you with issues :). Here is an example:

https://gist.github.com/Lucas-Steinmann/f900b3ef5636028d5917d4cb0183a291

>     the `early_stop_on` metric, specified in the initialization of the

Reviewer: ok good to know, let's fix this separately and later update the docs here

>     the `early_stop_on` metric, specified in the initialization of the

Author: I have started doing so with issue #3413 and PR #3414

    :class:`~pytorch_lightning.core.step_result.Result` object is used

>     and raise an error if it was not specified.

Author: You are correct. Strict checking is disabled with structured results. I've replaced it with ``and display a warning if it was not specified.``. I'll try to be more careful :)
